x-pack/plugins/ml/public/application/components/import_export_jobs/export_jobs_flyout/export_jobs_flyout.tsx

```diff
@@ -0,0 +1,263 @@
+  const [dfaJobIds, setDfaJobIds] = useState<string[]>([]);
+  const [selectedJobIds, setSelectedJobIds] = useState<string[]>([]);
+  const [exporting, setExporting] = useState(false);
+  const [selectedJobType, setSelectedJobType] = useState<JobType>(currentTab);
+  const { displayErrorToast, displaySuccessToast } = useMemo(
+    () => toastNotificationServiceProvider(toasts),
+    [toasts]
+  );
+
+  useEffect(() => {
+    setAdJobIds([]);
+    setSelectedJobIds([]);
+    setExporting(false);
+    setSelectedJobType(currentTab);
+
+    if (showFlyout) {
+      getJobs().then(({ jobs }) => {
+        setAdJobIds(jobs.map((j) => j.job_id));
+      });
+      getDataFrameAnalytics().then(({ data_frame_analytics: dataFrameAnalytics }) => {
+        setDfaJobIds(dataFrameAnalytics.map((j) => j.id));
+      });
+    }
+  }, [showFlyout]);
+
+  function toggleFlyout() {
+    setShowFlyout(!showFlyout);
+  }
+
+  async function onExport() {
+    setExporting(true);
+    const title = i18n.translate('xpack.ml.importExport.exportFlyout.exportDownloading', {
+      defaultMessage: 'Your file is downloading in the background',
+      values: { count: selectedJobIds.length },
+    });
+    displaySuccessToast(title);
+
+    try {
+      if (selectedJobType === 'anomaly-detector') {
+        await exportAnomalyDetectionJobs(selectedJobIds);
+      } else {
+        await exportDataframeAnalyticsJobs(selectedJobIds);
+      }
+
+      setExporting(false);
+      setShowFlyout(false);
+    } catch (error) {
+      const errorTitle = i18n.translate('xpack.ml.importExport.exportFlyout.exportError', {
+        defaultMessage: 'Could not export selected jobs',
+      });
+      displayErrorToast(error, errorTitle);
+      setExporting(false);
+    }
+  }
+
+  function toggleSelectedJob(checked: boolean, id: string) {
+    if (checked) {
+      setSelectedJobIds([...selectedJobIds, id]);
+    } else {
+      setSelectedJobIds(selectedJobIds.filter((id2) => id2 !== id));
+    }
+  }
+
+  function changeTab(jobType: JobType) {
+    setSelectedJobIds([]);
+    setSelectedJobType(jobType);
+  }
+
+  function onSelectAll() {
+    const ids = selectedJobType === 'anomaly-detector' ? adJobIds : dfaJobIds;
+    if (selectedJobIds.length === ids.length) {
+      setSelectedJobIds([]);
+    } else {
+      setSelectedJobIds([...ids]);
+    }
+  }
+
+  return (
+    <>
+      <FlyoutButton onClick={toggleFlyout} isDisabled={isDisabled} />
+
+      {showFlyout === true && isDisabled === false && (
+        <EuiFlyout onClose={() => setShowFlyout(false)} hideCloseButton size="s">
+          <EuiFlyoutHeader hasBorder>
+            <EuiTitle size="m">
+              <h2>
+                <FormattedMessage
+                  id="xpack.ml.importExport.exportFlyout.flyoutHeader"
+                  defaultMessage="Export jobs"
+                />
+              </h2>
+            </EuiTitle>
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody>
+            <EuiTabs size="s">
```

Comment:
If you make selections on one tab, and then switch to the other tab, the previous selections are silently discarded. 

![image](https://user-images.githubusercontent.com/7405507/126151143-9296d58b-ef84-451a-a78b-2d586f80667f.png)


It makes sense for the JSON file to only contain a single type of job, so I think the two options to address the issue of the user wondering why the export didn't contain the jobs they selected on the other tab are:
1. Only give the user the option exporting jobs of the type that were visible in the page when they hit the Export button, or
2. Add messaging when the user changes tabs in the flyout and they have jobs selected, that these selections will be lost when switching tabs

Thoughts?

Reply:
@pheyos from a usability point of view which of these behaviours do you think is best?
I'm in favour of adding a confirm message letting the user know that the current selection will be lost if they continue to switch job types (so option 2 above)

I think option 1 might cause confusion. Would it also suggest that they can only import jobs that match the current selection too?

Reply:
@jgowdyelastic I also think that a confirm modal on tab change is a good option here (but only if elements are selected in the current tab). Since there are no elements selected by default when opening the flyout, we wouldn't cause much click-overhead if the export type ic changed right away. And if elements on the initial tab have been selected it's fair to warn the user that this selection will be lost if the user continues. So a user who regularly runs through that workflow will quickly learn how it's working and won't see this extra modal a lot.

Reply:
Great, thanks @pheyos 
Yes, I was only talking about showing the modal if something is selected, otherwise there is no selection to lose.